cirq-core/cirq/circuits/circuit.py

```diff
@@ -1950,7 +1951,27 @@ def transform_qubits(
         with _compat.block_overlapping_deprecation(re.escape(_DEVICE_DEP_MESSAGE)):
             return Circuit(op_list, device=self._device if new_device is None else new_device)
 
-    def _prev_moment_available(self, op: 'cirq.Operation', end_moment_index: int) -> Optional[int]:
+    def prev_moment_available(
+        self, op: 'cirq.Operation', *, end_moment_index: Optional[int] = None
+    ) -> int:
+        """Finds the index of the previous (i.e. left most) moment which can accommodate `op`.
+
+        Note that, unlike `circuit.prev_moment_operating_on`, this method also takes care of
+        implicit dependencies between measurements and classically controlled operations (CCO)
+        that depend on the results of those measurements. Therefore, using this method, a CCO
+        `op` would not be allowed to move left past a measurement it depends upon.
+
+        Args:
+            op: Operation for which the previous moment that can accommodate it needs to be found.
+            end_moment_index: The moment index just after the starting point of the reverse search.
+                Defaults to the length of the list of moments.
+
+        Returns:
+            Index of the latest matching moment. Returns `end_moment_index` if no moment on left is
```

Comment:
This should be _earliest_ matching moment, right?

Reply:
done